### ISSUE TITLE
Space added when flattening columns on Snowflake

### ISSUE THREAD

**Issue by noel** [tconbeer/sqlfmt#103]
Given a column with the following sql
```
        _airbyte_data:"AccountId"::varchar as accountid,
```
sqlfmt will add additional spaces around `:` like so
```
            _airbyte_data : "AccountId"::varchar as accountid,
```

**Comment by tconbeer** [comment]
Thanks for the report, @noel!

I wasn't familiar with that syntax in Snowflake, but it's documented here (and it's not unique to the dialect as a prefix to field names): https://docs.snowflake.com/en/user-guide/json-basics-tutorial-flatten.html

It seems colons should only be surrounded by spaces when they are slice operators. We'll have to infer the context and then format appropriately

**Comment by tconbeer** [comment]
I should have mentioned, as a workaround, you can use `-- fmt: off` and `-- fmt: on` comments to surround the lines in your file that are causing the issue, so you can continue to use sqlfmt

**Comment by tconbeer** [comment]
Fixed by #106 -- this will be included in the 0.5.0 release this week.

**Comment by tconbeer** [comment]
0.5.0 has been released! Please upgrade for the patch
